File: ProyectosCSharp/SoftProg/SoftProgModel/Ventas/LineaOrdenVenta.cs
using SoftProgModel.Almacen;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgModel.Ventas
{
    public class LineaOrdenVenta
    {
        private int idLineaOrdenVenta;
        private Producto producto;
        private int cantidad;
        private double subtotal;
        private bool activa;

        public int IdLineaOrdenVenta { get => idLineaOrdenVenta; set => idLineaOrdenVenta = value; }
        public Producto Producto { get => producto; set => producto = value; }
        public int Cantidad { get => cantidad; set => cantidad = value; }
        public double Subtotal { get => subtotal; set => subtotal = value; }
        public bool Activa { get => activa; set => activa = value; }
    }
}

File: Arquitectura_REST_SOAP/SoftProgModel/src/main/java/pe/edu/pucp/softprog/ventas/model/LineaOrdenVenta.java
package pe.edu.pucp.softprog.ventas.model;

import pe.edu.pucp.softprog.almacen.model.Producto;

public class LineaOrdenVenta {
    private int idLineaOrdenVenta;
    private Producto producto;
    private int cantidad;
    private double subtotal;
    private boolean activo;

    public int getIdLineaOrdenVenta() {
        return idLineaOrdenVenta;
    }

    public void setIdLineaOrdenVenta(int idLineaOrdenVenta) {
        this.idLineaOrdenVenta = idLineaOrdenVenta;
    }

    public Producto getProducto() {
        return producto;
    }

    public void setProducto(Producto producto) {
        this.producto = producto;
    }

    public int getCantidad() {
        return cantidad;
    }

    public void setCantidad(int cantidad) {
        this.cantidad = cantidad;
    }

    public double getSubtotal() {
        return subtotal;
    }

    public void setSubtotal(double subtotal) {
        this.subtotal = subtotal;
    }

    public boolean isActivo() {
        return activo;
    }

    public void setActivo(boolean activo) {
        this.activo = activo;
    }
    
    public void calcularSubtotal(){
        subtotal = cantidad * producto.getPrecio();
    }
    
}